### DISCUSSION TITLE
Will retained edge functions ever be deleted?

### DISCUSSION THREAD

**Discussion by coltenkrauter** [cdklabs/cdk-nextjs#31] [Q&A] [answered]
In [nextjs-distribution.ts](https://github.com/cdklabs/cdk-nextjs/blob/734235143a5b9852d2883eca6bd5490ae94e9d73/src/nextjs-distribution.ts#L266C1-L267), the edge function uses `RemovalPolicy.RETAIN`.

```ts
// retain on delete b/c they take too long to delete resulting in stack failure
edgeFn.applyRemovalPolicy(RemovalPolicy.RETAIN);
```

Assuming that I delete the rest of the stack, does this mean the edge function/s will never be deleted automatically?

**Comment by bestickley** [comment, marked as answer]
@coltenkrauter, correct. I did this because in the future I want to have automated tests that automatically spin up and down cdk-nextjs apps and didn't want to deal CloudFormation Stack Deletion failures. But I'm open to changing this if you have good reason. Lambda@Edge functions will hopefully be optional in future. See: https://github.com/cdklabs/cdk-nextjs/issues/8

**Reply by coltenkrauter** [reply]
Thanks for this information, seems like a reasonable decision. Looking forward to #8 👍🏻
